### PR TITLE
New module: Add module to gather AWS CodeCommit SSH public key facts (cloud/amazon/codecommit_ssh_key_facts)

### PR DESCRIPTION
**Pull request by psharkey**
##### SUMMARY
codecommit_ssh_key_facts.py is a new module  for gathering AWS CodeCommit SSH key facts.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
codecommit_ssh_key_facts.py

##### ANSIBLE VERSION

```
ansible 2.2.1.0
  config file = /home/ansible.cfg
  configured module search path = Default w/o overrides
```

##### ADDITIONAL INFORMATION

References:
http://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_ssh-keys.html
http://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.get_ssh_public_key

##### EXAMPLE PLAYBOOK

```
---
- hosts: localhost
  tasks:

  - name: get SSH public key content for an IAM user
    codecommit_ssh_key_facts:
      user_name: "psharkey"
      encoding: 'SSH'
    register: ssh_key_facts

  - debug: var=ssh_key_facts
```

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/codecommit_ssh_key_facts.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+version_added: "2.4"
+author: "Pat Sharkey, (@psharkey)"
+options:
+  user_name:
+    description:
+      - The name of the user whose SSH public key will be returned
+    required: true
+  encoding:
+    description:
+      - The desired format for the returned key PEM | SSH
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/codecommit_ssh_key_facts.py]: Are these the only two valid options? 
If do please add as `choices:`  and add as choices to the arg spec. 

Of write 
`.. returned key, such as ssh or pen.`

**Review by gundalow — COMMENTED** [review]
1) name of the module doesn't seem to be online with the other AWS modules, though I'm not as familiar with them.

File: lib/ansible/modules/cloud/amazon/codecommit_ssh_key_facts.py (hunk 1 above)
Comment (psharkey) [reply lib/ansible/modules/cloud/amazon/codecommit_ssh_key_facts.py]: @gundalow Thanks for your comments. Regarding the name, AWS has a CodeCommit service (http://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_ssh-keys.html) and this module will gather the SSH public key(s) for all IAM users.  Do you have any suggestions for a different name - maybe `codecommit_ssh_facts` or `codecommit_facts` (with the caveat that only SSH information is currently supported)?

I'll work on adding `PEM` and `SSH` as the only valid choices to the encoding parameter. This is simply passed as a required parameter to the boto method `get_ssh_public_key` (http://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.get_ssh_public_key).

Comment (gundalow) [reply lib/ansible/modules/cloud/amazon/codecommit_ssh_key_facts.py]: Thanks for confirming, name sounds sensible to me them
http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#documentation-block

details `choices:`

File: lib/ansible/modules/cloud/amazon/codecommit_ssh_key_facts.py [hunk 2]
```diff
@@ -40,7 +40,7 @@
     required: true
   encoding:
     description:
-      - The desired format for the returned key PEM | SSH
+      - The desired format for the returned key which can be either PEM or SSH
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/codecommit_ssh_key_facts.py]:
```yaml
description:
  - The desired format for the returned key.
choices:
  - PEM
  - SSH
```

Comment (psharkey) [reply lib/ansible/modules/cloud/amazon/codecommit_ssh_key_facts.py]: @gundalow Thanks! - I think I got this right now...

File: lib/ansible/modules/cloud/amazon/codecommit_ssh_key_facts.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    user_name = module.params.get('user_name')
+    changed = False
+
+    args = {}
+    if user_name is not None:
+        args['UserName'] = user_name
+    try:
+        response = connection.list_ssh_public_keys(**args)
+    except ClientError as e:
+        module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
```
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/codecommit_ssh_key_facts.py:114]: In addition to the error message from boto, can you include a nice message like "ClientError while listing SSH public keys" for users? 

Also, please catch a generic exception and fail with the traceback, like `module.fail_json(msg="Could not list SSH public keys", exception=traceback.format_exc())`

**Review by ryansb — CHANGES_REQUESTED** [review]
Other than exception handling, looks great. Thanks for contributing!

**Comment by Akasurde** [comment]
Hi @psharkey, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.aws<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**